test/e2e/mirror_test.go

```diff
@@ -0,0 +1,160 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	config "github.com/openshift/api/config/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	batch "k8s.io/api/batch/v1"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/windows-machine-config-operator/controllers"
+	"github.com/openshift/windows-machine-config-operator/pkg/windows"
+)
+
+const (
+	testMirrorSetName               = "e2e-powershell-mirror"
+	upstreamPowershellImageLocation = "mcr.microsoft.com/powershell"
+	badPowershellImageLocation      = "test.registry.io/powershell"
+)
+
+func testImageMirroring(t *testing.T) {
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+
+	require.NoError(t, tc.loadExistingNodes(), "error getting the current Windows nodes in the cluster")
+
+	if !t.Run("Mirror settings applied to nodes", tc.testMirrorSettingsApplied) {
+		// No point in running the rest of the tests if settings aren't applied
+		return
+	}
+	t.Run("Workload using a mirrored container image", tc.testWorkloadWithMirroredImage)
+	t.Run("Mirror settings cleared from nodes", tc.testMirrorSettingsCleared)
```

Comment:
Please do this as a step in the deconfigure suite

Reply:
Hmm, if we moved this we would no longer be testing the registry-controller's reactivity to ITMS changes

Reply:
You would still be testing that if you remove the ITMS created for the tests to run properly and then wait for the registry directory to be cleared